packages/ifpack2/src/Ifpack2_BandedContainer_def.hpp

```diff
@@ -57,225 +57,242 @@
 
 namespace Ifpack2 {
 
+//Constructor that ContainerFactory and BlockRelaxation can use
 template<class MatrixType, class LocalScalarType>
-BandedContainer<MatrixType, LocalScalarType, true>::
+BandedContainer<MatrixType, LocalScalarType>::
 BandedContainer (const Teuchos::RCP<const row_matrix_type>& matrix,
                  const Teuchos::Array<Teuchos::Array<local_ordinal_type> >& partitions,
-                 const Teuchos::RCP<const import_type>& importer,
-                 int OverlapLevel,
-                 scalar_type DampingFactor) :
-  Container<MatrixType>(matrix, partitions, importer, OverlapLevel, DampingFactor),
-  ipiv_(this->partitions_.size()),
+                 const Teuchos::RCP<const import_type>&,
+                 bool pointIndexed) :
+  ContainerImpl<MatrixType, LocalScalarType>(matrix, partitions, pointIndexed),
+  ipiv_(this->blockRows_.size() * this->scalarsPerRow_),
   kl_(this->numBlocks_, -1),
   ku_(this->numBlocks_, -1),
-  scalars_(nullptr),
   scalarOffsets_(this->numBlocks_)
 {
+  TEUCHOS_TEST_FOR_EXCEPTION(!::Ifpack2::Details::LapackSupportsScalar<LocalScalarType>::value,
```

Comment:
@mhoemmen It turns out that Teuchos::LAPACK compiles fine with unsupported scalars, and throws a runtime error in every function in that case. I couldn't do a compile-time error in Ifpack2 because a downstream package instantiates this with Scalar = Sacado::Vector, but never actually uses it.